pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -741,16 +741,32 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 	return nil
 }
 
-func applyResource(u *unstructured.Unstructured, logger log.FieldLogger, applyFnMetricLabel string, applyFn func(obj []byte) (hiveresource.ApplyResult, error)) error {
-
-	startTime := time.Now()
-	labels := u.GetLabels()
+// addHiveLabels injects labels to help end-users and support see that a resource is managed by hive,
+// and which [Selector]SyncSet it originated from.
+func addHiveLabels(ssi *hivev1.SyncSetInstance, obj metav1.Object) {
+	labels := obj.GetLabels()
 	if labels == nil {
-		labels = make(map[string]string, 1)
+		labels = make(map[string]string, 2)
 	}
-	// Inject the hive managed annotation to help end-users see that a resource is managed by hive:
+	// Resource is managed by hive:
 	labels[constants.HiveManagedLabel] = "true"
-	u.SetLabels(labels)
+	// Which [Selector]SyncSet it originated from:
+	if ssi.Spec.SelectorSyncSetRef != nil {
+		labels[constants.SelectorSyncSetNameLabel] = ssi.Spec.SelectorSyncSetRef.Name
+	} else if ssi.Spec.SyncSetRef != nil {
+		labels[constants.SyncSetNameLabel] = ssi.Spec.SyncSetRef.Name
+	}
+	// TODO(efried): else panic? This is a developer error (forgot to add a condition for new ssi source type).
```

Comment:
I don't think we want to panic on some badly formatted SSI (or b/c we downgraded Hive for some reason), but at minimum we could log the fact that we've encountered an unexpected SSI.